isilon_create_directories.sh

```diff
@@ -54,54 +54,54 @@ function makedir() {
    if [ "z$1" == "z" ] ; then
       echo "ERROR -- function makedir needs directory as an argument"
    else
-      mkdir -p $1
+      mkdir -p "$1"
    fi
 }
 
 function fixperm() {
    if [ "z$1" == "z" ] ; then
       echo "ERROR -- function fixperm needs directory owner group perm as an argument"
    else
-      uid=$(getUserUid $2)
-      gid=$(getGroupGid $3)
-      chown $uid $1
-      chown :$gid $1
-      if [ "POSIX" == "y" ] ; then
-          chmod -D $1
+      uid="$(getUserUid "$2")"
+      gid="$(getGroupGid "$3")"
+      chown "$uid" "$1"
+      chown ":$gid" "$1"
+      if [ "$POSIX" == "y" ] ; then
+          chmod -D "$1"
       fi
-      chmod $4 $1
+      chmod "$4" "$1"
    fi
 }
 
 function getHdfsRoot() {
     local hdfsroot
     #Check for Version to process correct syntax - isirad
-    if [ "`isi version|cut -c 15`" -lt 8 ]; then
-      hdfsroot=$(isi zone zones view $1 | grep "HDFS Root Directory:" | cut -f2 -d :)
+    if [ "$(isi version|cut -c 15)" -lt 8 ]; then
+      hdfsroot="$(isi zone zones view "$1" | grep "HDFS Root Directory:" | cut -f2 -d :)"
     else
-      hdfsroot=$(isi hdfs settings view --zone=$1 | grep "Root Directory:" | cut -f2 -d :)
+      hdfsroot="$(isi hdfs settings view --zone="$1" | grep "Root Directory:" | cut -f2 -d :)"
     fi
-    echo $hdfsroot
+    echo "$hdfsroot"
```

Comment:
`getHdfsRoot` needs to strip whitespace:
```
dtucker-4x77q2i-1# bash /ifs/isilon_create_directories.sh --dist cdh --zone System --append-cluster-name somecluster1
Info: Hadoop distribution:  cdh
Info: will use users in zone:  System
Info: will add clustername to end of usernames: -somecluster1
Info: HDFS root dir is  /ifs
FATAL:  HDFS root  /ifs does not exist!
dtucker-4x77q2i-1# bash ~/isilon_create_directories.sh --dist cdh --zone System --append-cluster-name somecluster1
Info: Hadoop distribution:  cdh
Info: will use users in zone:  System
Info: will add clustername to end of usernames: -somecluster1
Info: HDFS root dir is /ifs
##################################################################################
```
- `~/isilon_create_directories.sh` is from master.